### PR TITLE
morello: add sysctl for logging user cheri exceptions

### PR DESCRIPTION
**Pull request by brettferdosi**
Set machdep.log_user_cheri_exceptions to enable logging.

### CODE REVIEW

File: sys/arm64/arm64/trap.c [hunk 1]
```diff
@@ -273,7 +280,18 @@ cap_abort(struct thread *td, struct trapframe *frame, uint64_t esr,
...
+	}
+
+	if (log_user_cheri_exceptions) {
+		printf("pid %d tid %d (%s) uid %d: capability abort, %s\n",
+		    td->td_proc->p_pid, td->td_tid, td->td_proc->p_comm,
+		    td->td_ucred->cr_uid,
+		    cheri_fsc_string(esr & ISS_DATA_DFSC_MASK));
+		print_registers(frame);
+		printf(" far: %16lx\n", far);
+		printf(" esr:         %.8lx\n", esr);
```
Comment (brooksdavis) [inline sys/arm64/arm64/trap.c:294]: Why the differing indentation?

Comment (jrtc27) [reply sys/arm64/arm64/trap.c:294]: Convention taken from elsewhere so everything is right-aligned

Comment (brooksdavis) [reply sys/arm64/arm64/trap.c:294]:
```suggestion
		printf(" far: %16lx\n", far);
		printf(" esr: % 16.8lx\n", esr);
```
Maybe make that part of the format?

Comment (jrtc27) [reply sys/arm64/arm64/trap.c:294]: Perhaps better, but then inconsistent with the other 9 instances of this in the file.

Comment (brooksdavis) [reply sys/arm64/arm64/trap.c:294]: Maybe should tidy these in a later commit and upstream the others.

**Review by brooksdavis — APPROVED** [review]

**Pull request merged**